packages/tre/src/plugins/shared/gateway.ts

```diff
@@ -25,21 +30,39 @@ export class GatewayFactory<Gateway> {
   ) {}
 
   #storage(namespace: string, persist: Persistence): Storage {
+    // If persistence is disabled, use memory storage
     if (persist === undefined || persist === false) {
       let storage = this.#memoryStorages.get(namespace);
       if (storage !== undefined) return storage;
       this.#memoryStorages.set(namespace, (storage = new MemoryStorage()));
       return storage;
     }
 
-    const sanitised = sanitisePath(namespace);
+    // Sanitise namespace to make it file-system safe
+    const sanitisedNamespace = sanitisePath(namespace);
+
+    // If `persist` looks like a URL, parse it
+    if (persist !== true && URL_REGEXP.test(persist)) {
```

Comment:
Is the `URL_REGEXP` necessary? What about just catching the `TypeError` from `new URL()` to handle invalid URLs?